logstash-core/lib/logstash/agent.rb

```diff
@@ -23,19 +29,28 @@ class LogStash::Agent
   def initialize(params)
     @logger = params[:logger]
     @auto_reload = params[:auto_reload]
-    @pipelines = {}
+    @debug  = params.fetch(:debug, false)
```

Comment:
I guess this will need rebase with other changes proposed at https://github.com/elastic/logstash/pull/4677